ios/HybridImageFactory.swift

```diff
@@ -21,14 +22,102 @@ class HybridImageFactory: HybridImageFactorySpec {
     guard let url = URL(string: urlString) else {
       throw RuntimeError.error(withMessage: "URL string \"\(urlString)\" is not a valid URL!")
     }
-
+    
     return Promise.async {
       let webImageOptions = options?.toSDWebImageOptions() ?? []
       let uiImage = try await SDWebImageManager.shared.loadImage(with: url, options: webImageOptions)
       return HybridImage(uiImage: uiImage)
     }
   }
   
+  /**
+   * Load Image from URL
+   */
+  func loadFromAssetAsync(assetId: String, options: AssetImageLoadOptions?) throws -> Promise<any HybridImageSpec> {
+    let assets = PHAsset.fetchAssets(
+      withLocalIdentifiers: [assetId],
+      options: nil
+    )
+    guard let asset = assets.firstObject else {
+      throw NSError(
+        domain: "TurboImageView",
+        code: 404,
+        userInfo: [NSLocalizedDescriptionKey: "Asset not found"]
+      )
+    }
+    
+    return Promise.async {
+      return try await withCheckedThrowingContinuation { continuation in
+        let requestOptions = PHImageRequestOptions()
+        requestOptions.version = .current
+        requestOptions.deliveryMode = .highQualityFormat
+        requestOptions.isNetworkAccessAllowed = true
+        
+        
+        if let size = options?.size {
+          let contentMode: PHImageContentMode
+          if let aspectFit = options?.aspectFit {
+            switch aspectFit {
+            case .fit:
+              contentMode = .aspectFit
+            case .fill:
+              contentMode = .aspectFill
+            }
+          } else {
+            contentMode = .default
+          }
```

Comment:
Create an extension on `AspectFit` that gives you a `PHImageContentMode`. Same as above, `AspectFit+toPHImageContentMode()` or something. No reason to do this inline